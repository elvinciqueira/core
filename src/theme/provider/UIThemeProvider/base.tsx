import React from "react";
import { ThemeProvider as ThemeProviderNative } from "styled-components/native";
import { ThemeProvider as ThemeProviderWeb } from "styled-components";
import createTheme from "../../../theme/createTheme";
import { Platforms } from "../../../theme/types/Platforms";

interface UIThemeProviderProps {
  platform: Platforms;
  children: React.ReactNode;
}
export default function UIThemeProvider({
  children,
  platform,
}: UIThemeProviderProps): JSX.Element {
  const theme = createTheme("light", platform);

  if (platform === "mobile") {
    return <ThemeProviderNative theme={theme}>{children}</ThemeProviderNative>;
  }

  return <ThemeProviderWeb theme={theme}>{children}</ThemeProviderWeb>;
}
