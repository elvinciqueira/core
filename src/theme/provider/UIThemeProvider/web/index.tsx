import React from 'react';
import UIThemeProvider from '../base';
import GlobalStyle from '@src/theme/web/GlobalStyle';

interface UIThemeProviderWebProps {
  children: React.ReactNode;
}
export default function UIThemeProviderWeb({
  children,
}: UIThemeProviderWebProps) {
  return (
    <UIThemeProvider platform="web">
      <GlobalStyle />
      {children}
    </UIThemeProvider>
  );
}
