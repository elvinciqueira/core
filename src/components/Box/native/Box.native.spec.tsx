import React from "react";
import renderer from "react-test-renderer";
import UIThemeProvider from "../../../theme/provider/UIThemeProvider/base";
import Box from ".";

describe("Box", () => {
  test("renders correctly", () => {
    const tree = renderer
      .create(
        <UIThemeProvider platform="mobile">
          <Box as={{ web: "main", mobile: "ScrollView" }}>Native box</Box>
        </UIThemeProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
