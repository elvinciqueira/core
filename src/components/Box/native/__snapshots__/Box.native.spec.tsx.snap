// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": undefined,
      "flexGrow": 1,
      "justifyContent": undefined,
    }
  }
  style={
    Array [
      Object {
        "width": "100%",
      },
    ]
  }
>
  <View>
    Native box
  </View>
</RCTScrollView>
`;
